TCC/Models/MemorialDeCalculo.swift

```diff
@@ -0,0 +1,259 @@
+        return soma / Double(numeros.count)
+    }
+    
+    func mesesAtendidosParcialmente(dadosMeses: [DadosConsumoMes]) -> Double {
+        let count = dadosMeses.filter { $0.potencialDeAtendimentoDaDemanda > 0.01 && $0.potencialDeAtendimentoDaDemanda < 1 }.count
+        return Double(count) / Double(precipitacaoMensal.count / 12)
+    }
+    
+    func mesesAtendidosCompletamente(dadosMeses: [DadosConsumoMes]) -> Double {
+        let count = dadosMeses.filter { $0.potencialDeAtendimentoDaDemanda == 1 }.count
+        return Double(count) / Double(precipitacaoMensal.count / 12)
+    }
+    
+    func mesesNaoAtendidos(dadosMeses: [DadosConsumoMes]) -> Double {
+        let count = dadosMeses.filter { $0.potencialDeAtendimentoDaDemanda < 0.05 }.count
+        return Double(count) / Double(precipitacaoMensal.count / 12)
+    }
+    
+    func outrosCustos() -> Double {
+        return custoBomba + custoClorador + custoTubulacao
+    }
+    
+    func investimentoInicial(volume: Double) -> Double {
+        return outrosCustos() + capex.custoTotalReservatorios(volume: volume)
+    }
+    
+    //L
+    func economiaDeAguaAnual(potencialMedioDeAtendimentoDaDemanda: Double, demandaMensal: Double) -> Double {
+        return potencialMedioDeAtendimentoDaDemanda * demandaMensal * 12
+    }
+    
+    func economiaFinanceiraAnual(potencialMedioDeAtendimentoDaDemanda: Double, demandaMensal: Double) -> Double {
+        let economiaDeAguaAnualEmLitros = economiaDeAguaAnual(potencialMedioDeAtendimentoDaDemanda: potencialMedioDeAtendimentoDaDemanda, demandaMensal: demandaMensal)
+        let economiaDeAguaMensalEmMetrosCubicos = economiaDeAguaAnualEmLitros / (1000 * 12) // Convertendo para m³
+        return custoDaAgua(consumo: economiaDeAguaMensalEmMetrosCubicos) * 12
+    }
+
+    
+    func tempoDeRetorno(volume: Double, potencialMedioDeAtendimentoDaDemanda: Double, demandaMensal: Double) -> Double {
+        let economiaEfetivaAnual = economiaFinanceiraAnual(potencialMedioDeAtendimentoDaDemanda: potencialMedioDeAtendimentoDaDemanda, demandaMensal: demandaMensal) - opex
+        return investimentoInicial(volume: volume) / economiaEfetivaAnual
+    }
+    
+    // R$/m3
+    func custoDaAgua(consumo: Double) -> Double {
+        if consumo <= 10 {
+            return (45.94 + 36.75 + 19.76) * consumo
+        } else if consumo <= 15 {
+            return (8.53 + 6.83 + 3.65) * consumo - 39.36 - 31.55 - 16.74
+        } else if consumo <= 20 {
+            return (8.72 + 6.98 + 3.75) * consumo - 42.41 - 33.8 - 18.24
+        } else if consumo <= 25 {
+            return (8.94 + 7.12 + 3.86) * consumo - 46.41 - 36.6 - 20.44
+        } else if consumo <= 30 {
+            return (10.98 + 8.77 + 4.71) * consumo - 97.61 - 77.85 - 41.69
+        } else if consumo <= 50 {
+            return (11.68 + 9.34 + 5.03) * consumo - 118.61 - 94.95 - 51.29
+        } else {
+            return (17.93 + 14.3 + 7.7) * consumo - 413.11 - 342.25 - 184.79
+        }
```

Comment:
Se for possível agregar estes 6 parâmetros em uma struct significativa podia se pensar em criar um array com elas  (algo como parâmetros + valor máximo (10,15,20...) e apenas percorrer esse array retornando quando chegar ao valor correto. Isso diminuiria as linhas de código aqui e aumentaria flexibilidade no caso de mais faixas...